backend/src/main/java/fk/prof/backend/worker/BackendDaemon.java

```diff
@@ -0,0 +1,198 @@
+
+      //TODO: load = 0.5 hard-coded right now. Replace this with dynamic load computation in future
+      float load = 0.5f;
+
+      try {
+        httpClient.requestAsync(
+            HttpMethod.POST,
+            leaderIPAddress,
+            leaderHttpPort,
+            ApiPathConstants.LEADER_POST_LOAD,
+            ProtoUtil.buildBufferFromProto(
+                BackendDTO.LoadReportRequest.newBuilder()
+                    .setIp(ipAddress)
+                    .setPort(backendHttpPort)
+                    .setLoad(load)
+                    .setCurrTick(loadTickCounter++)
+                    .build()))
+            .setHandler(ar -> {
+              if(ar.succeeded()) {
+                if(ar.result().getStatusCode() == 200) {
+                  try {
+                    Recorder.ProcessGroups assignedProcessGroups = ProtoUtil.buildProtoFromBuffer(Recorder.ProcessGroups.parser(), ar.result().getResponse());
+                    processGroupAssociationStore.updateProcessGroupAssociations(assignedProcessGroups, (processGroupDetail, processGroupAssociationResult) -> {
+                      switch (processGroupAssociationResult) {
+                        case ADDED:
+                          this.aggregationWindowPlannerStore.associateAggregationWindowPlannerIfAbsent(processGroupDetail);
+                          break;
+                        case REMOVED:
+                          this.aggregationWindowPlannerStore.deAssociateAggregationWindowPlanner(processGroupDetail.getProcessGroup());
+                          break;
+                      }
+                    });
+                  } catch (Exception ex) {
+                    logger.error("Error parsing response returned by leader when reporting load");
+                  }
+                } else {
+                  logger.error("Non OK status returned by leader when reporting load, status=" + ar.result().getStatusCode());
+                }
+              } else {
+                logger.error("Error when reporting load to leader", ar.cause());
+              }
+              setupTimerForReportingLoad();
+            });
+      } catch (IOException ex) {
+        logger.error("Error building load request body", ex);
+        setupTimerForReportingLoad();
+      }
+    } else {
+      logger.debug("Not reporting load because leader is unknown");
+      setupTimerForReportingLoad();
+    }
+  }
+
+  private void setupTimerForReportingLoad() {
+    vertx.setTimer(configManager.getLoadReportIntervalInSeconds() * 1000, timerId -> postLoadToLeader());
+  }
+
+  private Future<BackendDTO.WorkProfile> getWorkFromLeader(Recorder.ProcessGroup processGroup) {
+    Future<BackendDTO.WorkProfile> result = Future.future();
+    String leaderIPAddress;
+    if((leaderIPAddress = leaderReadContext.getLeaderIPAddress()) != null) {
+      try {
+        String requestPath = new StringBuilder(ApiPathConstants.LEADER_GET_WORK)
```

Comment:
Method for this path generation and a class for leader-bound http-client will help reduce this repetitive code. Eg. for http-client, host, port etc can be constructor params. May be separate wrapper objects over httpClient that encap path and verb too?